extensions/amp-story/1.0/amp-story-page.js

```diff
@@ -1757,29 +1795,140 @@ export class AmpStoryPage extends AMP.BaseElement {
       );
 
+      const ctaChipEl = this.openAttachmentEl_.querySelector(
+        '.i-amphtml-story-page-attachment-outlink-chip'
+      );
       const textEl = this.openAttachmentEl_.querySelector(
-        '.i-amphtml-story-page-open-attachment-label'
+        '.i-amphtml-story-page-attachment-outlink-label'
+      );
+      const ctaImgEl = this.openAttachmentEl_.querySelector(
+        '.i-amphtml-story-page-attachment-outlink-image'
+      );
+      const barLeftEl = this.openAttachmentEl_.querySelector(
+        '.i-amphtml-story-page-open-attachment-outlink-bar-left'
+      );
+      const barRightEl = this.openAttachmentEl_.querySelector(
+        '.i-amphtml-story-page-open-attachment-outlink-bar-right'
       );
 
-      const openLabelAttr = attachmentEl.getAttribute('data-cta-text');
+      const openLabelAttr = attachmentEl.getAttribute('cta-text');
       const openLabel =
         (openLabelAttr && openLabelAttr.trim()) ||
         getLocalizationService(this.element).getLocalizedString(
           LocalizedStringId.AMP_STORY_PAGE_ATTACHMENT_OPEN_LABEL
         );
 
       this.mutateElement(() => {
-        textEl.textContent = openLabel;
+        const theme = attachmentEl.getAttribute('theme');
+        const ctaImgAttr = attachmentEl.getAttribute('cta-img');
+        if (ctaImgAttr !== 'none') {
+          setImportantStyles(textEl, {
+            'padding-left': '10px',
+          });
+          if (!ctaImgAttr) {
+            if (theme && AttachmentTheme.DARK === theme.toLowerCase()) {
+              ctaImgEl.classList.add(
+                'i-amphtml-story-page-attachment-outlink-dark-link-image'
+              );
+            } else {
+              ctaImgEl.classList.add(
+                'i-amphtml-story-page-attachment-outlink-light-link-image'
+              );
+            }
+          } else {
+            setImportantStyles(ctaImgEl, {
+              height: '24px',
+              width: '24px',
+              'background-image': 'url(' + ctaImgAttr + ')',
+            });
+          }
+        }
+
+        const ctaAccentColor = attachmentEl.getAttribute('cta-accent-color');
+        if (theme && AttachmentTheme.DARK === theme.toLowerCase()) {
+          setImportantStyles(ctaChipEl, {
+            color: 'white',
+            background: ctaAccentColor ? ctaAccentColor : 'black',
+          });
+          setImportantStyles(barRightEl, {
+            background: ctaAccentColor ? ctaAccentColor : 'black',
+          });
+          setImportantStyles(barLeftEl, {
+            background: ctaAccentColor ? ctaAccentColor : 'black',
+          });
+        } else {
+          setImportantStyles(ctaChipEl, {
+            color: ctaAccentColor ? ctaAccentColor : 'black',
+            background: 'white',
+          });
+          setImportantStyles(barRightEl, {
+            background: 'white',
+          });
+          setImportantStyles(barLeftEl, {
+            background: 'white',
+          });
+        }
```

Comment:
This color styling logic can be handled using [css variables](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties).
In this way we can remove lots of queryselectors and if statements.

A default value like `--cta-accent-color` can be set on the parent element `i-amphtml-story-page-open-attachment`. and all children styles can be set with that CSS variable.

You can reference an example of this pattern in the interactive components:
https://github.com/ampproject/amphtml/blob/master/extensions/amp-story/1.0/amp-story.css#L902-L904

You can see how it's overridden in the template here:
https://github.com/ampproject/amphtml/blob/master/examples/amp-story/interactive_polls.html#L62

The "theme" styling can also follow this pattern.

Reply:
the `theme` and style attribute might need to be propagated to the wrapper element since it's not a descendant. 
It can follow this pattern:

```
      const theme = attachmentEl.getAttribute('theme');
      if (theme && AttachmentTheme.DARK === theme.toLowerCase()) {
        this.mutateElement(() =>
          this.openAttachmentEl_.setAttribute('theme', 'dark')
        );
      }

      const ctaAccentColor = attachmentEl.style.getPropertyValue(
        '--cta-accent-color'
      );
      if (ctaAccentColor) {
        this.mutateElement(() =>
          this.openAttachmentEl_.style.setProperty(
            '--cta-accent-color',
            ctaAccentColor
          )
        );
      }
```